[ace] Added TAO as feature

bold84 opened the pull request:
Added CMake files to find ACE and TAO

bold84 committed:
ace-tao 6.5.6

parent 1ebb733d78d592b597e46bc7c5a491e646023ed3
author Benjamin Oldenburg <benjamin.oldenburg@ordis.co.th> 1570361425 +0700
committer Benjamin Oldenburg <benjamin.oldenburg@ordis.co.th> 1570404986 +0700

Added TAO to ACE port

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=8468) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=8468) before we can accept your contribution.<br/><br/>:x: ordisbold [sign now](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=8468)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vcpkg?pullRequest=8468) it.</sub>

bold84 committed:
Enabled building for ARM, ARM64, UWP, Darwin

bold84 committed:
fix CMake script

ordisbold commented:
I won't make any further changes. All builds succeed.

bold84 committed:
Merge branch 'master' into vcpkg-ace-port

cbezault commented:
Why are you providing find modules? You should be providing config files if anything.

ordisbold commented:
I wasn't aware of that. I'll have a look at the documentation. I'd
appreciate if you gave me a hint though.

Curtis J Bezault <notifications@github.com> schrieb am Di., 8. Okt. 2019,
05:44:

> Why are you providing find modules? You should be providing config files
> if anything.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/vcpkg/pull/8468?email_source=notifications&email_token=AFBD2MO6N3KKBUFYTJDD7J3QNO3TPA5CNFSM4I54EBP2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEASBJNY#issuecomment-539235511>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFBD2MNISUYIGKWNNP56ZW3QNO3TPANCNFSM4I54EBPQ>
> .
>

Voskrese commented:
hmm... error

```
\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(209,5): error MSB6006: "cmd.exe" завершилась с кодом -1073741515. [E:\tools\vcpkg\buildtrees\ace\src\ACE_wrappers\tao\tao\TransportCurrent\TC_IIOP_Idl.vcxproj]
  Invoking "E:\tools\vcpkg\buildtrees\ace\src\ACE_wrappers\bin\tao_idl on ZIOP.pidl"
```


```
      <Command Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">setlocal&#x0D;&#x0A;PATH=%PATH%;E:\tools\vcpkg\buildtrees\ace\src\ACE_wrappers\lib&#x0D;&#x0A;cmd /c &quot;E:\tools\vcpkg\buildtrees\ace\src\ACE_wrappers\bin\tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -IE:\tools\vcpkg\buildtrees\ace\src\ACE_wrappers\tao -Wb,versioning_begin=TAO_BEGIN_VERSIONED_NAMESPACE_DECL -Wb,versioning_end=TAO_END_VERSIONED_NAMESPACE_DECL -Wb,export_include=tao/TransportCurrent/Transport_Current_Export.h -Wb,export_macro=TAO_Transport_Current_Export -iC tao/TransportCurrent -I./TransportCurrent -SS -Sci &quot;TC_IIOP.idl&quot;&quot;&#x0D;&#x0A;endlocal</Command>
      <Message Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">Invoking "E:\tools\vcpkg\buildtrees\ace\src\ACE_wrappers\bin\tao_idl on TC_IIOP.idl"</Message>
```

tag `&quot;` causes an error or does not find the file `TC_IIOP.idl or ace/pre.h or ace/post.h`

and such everywhere, where present `<CustomBuild`

bold84 committed:
Now not copying the CMake Find files anymore.

Corrected the installation location for TAO_IDL

ordisbold commented:
> hmm... error
> 
I originally started this port with another name a few weeks ago. While merging it to this ACE port, I forgot to change the target directory for the executables (i.e. tao_idl) from ace-tao to ace. That's fixed now.

Voskrese commented on ports/ace/portfile.cmake:
```diff
@@ -1,181 +1,315 @@
 include(vcpkg_common_functions)
 
-# Don't change to vcpkg_from_github! This points to a release and not an archive
+set(ACE_ROOT ${CURRENT_BUILDTREES_DIR}/src/ACE_wrappers)
```
```suggestion
set(ACE_ROOT ${CURRENT_BUILDTREES_DIR}/src/ACE_wrappers)
```

https://github.com/DOCGroup/ACE_TAO/blob/57c52c1183d8d5e11e725fc9886263dfca11f22c/azure-pipelines.yml#L2

? MPC_ROOT
https://github.com/DOCGroup/ACE_TAO/blob/57c52c1183d8d5e11e725fc9886263dfca11f22c/azure-pipelines.yml#L4

ordisbold replied:
@Voskrese 

**ACE_ROOT & MPC_ROOT**
The folder structure of the ACE_TAO git repository is different from the one in the archive. 

**default.features**
Yes. I will change that. :-)

JackBoosY replied:
Has this requirement been resolved?

Voskrese commented:
can immediately add?
`CIAO_ROOT, DANCE_ROOT, DDS_ROOT, XSC_ROOT`

cbezault commented:
In general find modules are written by the consumers of cmake libraries. Config files are created by the consumed library for the benefit of the consumers. Both define targets and all other variables necessary to consume a library.

Voskrese commented:
`vcpkg install ace[*]:x64-windows`

`    ace[core,qt5,ssl,tao,tao-bzip2,tao-zlib,wchar,xml]:x64-windows`

error
`?strdup@ACE_OS@@YAPEADPEBD@Z`
`Точка входа в процедуру ?strdup@ACE_OS@@YAPEADPEBD@Z не найдена в библиотеке DLL E:\tools\vcpkg\buildtrees\ace\src\ACE_wrappers\lib\TAO_IDL_BE.dll.`
`
E:/tools/vcpkg/downloads/tools/perl/perl/bin/perl.exe E:/tools/vcpkg/buildtrees/ace/src/ACE_wrappers/bin/mwc.pl -type vs2017 TAO_ACE.mwc -features stl=1,ace_for_tao=0,ace_inline=0,uses_wchar=1,xml=1,ssl=1,qt5=1,bzip2=1,zlib=0 -workers 5 -use_env -expand_vars`

`ace[core,qt5,ssl,tao,tao-bzip2,tao-zlib,wchar,xml]`

?
**,zlib=0,ace_for_tao=0,ace_inline=0**
`-features stl=1,ace_for_tao=0,ace_inline=0,uses_wchar=1,xml=1,ssl=1,qt5=1,bzip2=1,zlib=0`

ordisbold commented:
> **other error**
> 

@Voskrese 
PS C:\vcpkg> .\vcpkg.exe install ace[*]
The following packages will be built and installed:
    ace[core,qt5,ssl,tao,tao-bzip2,tao-zlib,wchar,xml]:x64-windows
Starting package 1/1: ace:x64-windows
Building package ace[core,qt5,ssl,tao,tao-bzip2,tao-zlib,wchar,xml]:x64-windows...
Warning: abi keys are missing values:
    bzip2
    openssl
    qt5-base
    zlib

-- Using cached C:/vcpkg/downloads/ACE+TAO-src-6.5.6.tar.gz
-- Extracting source C:/vcpkg/downloads/ACE+TAO-src-6.5.6.tar.gz
-- Applying patch C:/vcpkg/ports/ace/qtcoreapplication.patch
-- Applying patch C:/vcpkg/ports/ace/bzip2.patch
-- Applying patch C:/vcpkg/ports/ace/mpc-arm.patch
-- Applying patch C:/vcpkg/ports/ace/stacktrace-arm.patch
-- Applying patch C:/vcpkg/ports/ace/arm_prevent_amd64_definition.patch
,uses_wchar=1,xml=1,ssl=1,qt5=1,bzip2=1,zlib=1
-- Building C:/vcpkg/buildtrees/ace/src/ACE_wrappers/tao/TAO_ACE.sln for Release

Voskrese commented:
is guilty, means at me remained everything in a cache, files were not reconfigured
will look further

jwillemsen commented on ports/ace/CONTROL:
```diff
@@ -1,14 +1,33 @@
-Homepage: https://www.dre.vanderbilt.edu/~schmidt/ACE.html
 Description: The ADAPTIVE Communication Environment
+Homepage: http://www.dre.vanderbilt.edu/~schmidt/TAO.html
+Maintainer: Benjamin Oldenburg <benjamin.oldenburg@ordis.co.th>
```
Maintainer is not a field they support as far as I can find, see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md

jwillemsen commented on ports/ace/bzip2.patch:
```diff
@@ -0,0 +1,11 @@
+--- a/MPC/config/bzip2.mpb	2019-10-06 11:14:51.164094600 +0700
++++ b/MPC/config/bzip2.mpb	2019-10-06 11:15:08.269289500 +0700
+@@ -5,8 +5,6 @@
```
Could you create a PR for MPC https://github.com/DOCGroup/MPC

ordisbold replied:
Yes... I've moved our office. I expect to be done tonight and can take care about this and the other PRs on the weekend.

jwillemsen commented on ports/ace/CONTROL:
```diff
@@ -1,14 +1,33 @@
 Version: 6.5.6
-Homepage: https://www.dre.vanderbilt.edu/~schmidt/ACE.html
 Description: The ADAPTIVE Communication Environment
+Homepage: http://www.dre.vanderbilt.edu/~schmidt/TAO.html
```
Please use https link

ordisbold replied:
Done

JackBoosY commented:
/azp run

jwillemsen commented:
We also have https://github.com/microsoft/vcpkg/pull/9016 as another PR in ace

jwillemsen commented:
See https://github.com/microsoft/vcpkg/pull/9074 for the new ACE release which has some of the patches integrated

jwillemsen commented:
@ordisbold Any plans to finish the TAO feature soon?

JackBoosY commented:
@ordisbold Ace in vcpkg has been updated to 6.5.7, does your change still work?

ordisbold commented:
> @ordisbold Any plans to finish the TAO feature soon?

Not at the moment. I got too many other things on my plate and it has proven to be time intensive to get it right for all CI targets.

PhoebeHui commented on ports/ace/CONTROL:
```diff
@@ -1,14 +1,33 @@
 Source: ace
 Version: 6.5.6
```
Could you please update Version to ' 6.5.6-1'?

PhoebeHui commented on ports/ace/CONTROL:
```diff
@@ -1,14 +1,33 @@
+
+Feature: tao-bzip2
+Build-Depends: bzip2, ace[tao]
+Description: Enable bzip2 support for TAO
```
Have you test these features locally?

ordisbold replied:
As in building a tao based application that uses compression with these libraries? No, of course I have not.

PhoebeHui commented on ports/ace/portfile.cmake:
```diff
@@ -1,181 +1,315 @@
+
 include(vcpkg_common_functions)
```
Could you please remove 'include(vcpkg_common_functions)', this is deprecated.

JackBoosY commented:
Please resolve these conflicts.

Thanks.

jwillemsen commented:
I think we first should get https://github.com/microsoft/vcpkg/pull/9016 and after that resolve the conflicts with this one, seems there is some overlap

jwillemsen commented:
#9016 got merged

jwillemsen commented:
new ACE/TAO release, see https://github.com/microsoft/vcpkg/pull/10003

bold84 committed:
Removed deprecated vcpkg_common_functions inclusion

Updated to TAO 6.5.8
Removed redundant patch files

bold84 committed:
changed the feature names (removed tao-)

bold84 committed:
Updated version in CONTROL file, removed Maintainer string and replaced Homepage protocol to https.

bold84 committed:
Merge remote-tracking branch 'upstream/master' into vcpkg-ace-port

# Conflicts:
#	ports/ace/CONTROL
#	ports/ace/portfile.cmake

bold84 committed:
temporarily added .gitlab-ci.yml

bold84 committed:
Update .gitlab-ci.yml

bold84 committed:
openssl11=1

bold84 committed:
Merge branch 'vcpkg-ace-port' of http://gitlab.intra.ordis.co.th/ordis/vcpkg into vcpkg-ace-port

JackBoosY commented:
Please remove the following content in _VCPKG_PATH/scripts/ci.baseline.txt_:
```
ace:x64-uwp=fail
ace:arm64-windows=fail
ace:arm-uwp=fail
```

jwillemsen commented:
Are all patches still required with TAO 2.5.8? Some got integrated already as far as I remember.

ordisbold commented:
> Are all patches still required with TAO 2.5.8? Some got integrated already as far as I remember.

I did remove them, just two were still necessary. I will look into it again

bold84 committed:
Removed entries from ci.ci.baseline.txt

bold84 committed:
removed .gitlab-ci.yml

ordisbold commented:
Have a look at 

** Test Modified Ports **

for OSX. 

The build fails, but the status of the job is "succeeded". Something wrong there!?

jwillemsen commented:
Looks macosX is allowed to fail because it didn't work before, from the log I can find the following, no idea why make fails, I can't find the detailed log they refer to

```
2020-04-01T23:57:41.0768120Z Could not locate cached archive: /Volumes/data/work/1/s/archives/1d/1d6d510a49f5d1baaf7084deeeffb6f7b1ea6287.zip
2020-04-02T00:00:23.6257350Z -- Downloading https://github.com/DOCGroup/ACE_TAO/releases/download/ACE%2BTAO-6_5_8/ACE+TAO-src-6.5.8.tar.gz...
2020-04-02T00:00:23.6259490Z -- Extracting source /Volumes/data/work/1/s/downloads/ACE+TAO-src-6.5.8.tar.gz
2020-04-02T00:00:23.6260260Z -- Applying patch /Volumes/data/work/1/s/ports/ace/arm_prevent_amd64_definition.patch
2020-04-02T00:00:23.6260870Z -- Applying patch /Volumes/data/work/1/s/ports/ace/bzip2.patch
2020-04-02T00:00:23.6261070Z CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
2020-04-02T00:00:23.6261220Z     Command failed: make
2020-04-02T00:00:23.6261380Z     Working Directory: /Volumes/data/work/1/s/buildtrees/ace/src/ACE_wrappers
2020-04-02T00:00:23.6273580Z     Error code: 2
2020-04-02T00:00:23.6274010Z     See logs for more information:
2020-04-02T00:00:23.6275600Z       /Volumes/data/work/1/s/buildtrees/ace/make-x64-osx-out.log
2020-04-02T00:00:23.6276180Z       /Volumes/data/work/1/s/buildtrees/ace/make-x64-osx-err.log
2020-04-02T00:00:23.6276710Z 
2020-04-02T00:00:23.6277000Z Call Stack (most recent call first):
2020-04-02T00:00:23.6278280Z   ports/ace/portfile.cmake:183 (vcpkg_execute_required_process)
2020-04-02T00:00:23.6278450Z   scripts/ports.cmake:90 (include)
2020-04-02T00:00:23.6278700Z 
2020-04-02T00:00:23.6278860Z 
2020-04-02T00:00:24.8916680Z Error: Building package ace:x64-osx failed with: BUILD_FAILED
2020-04-02T00:00:24.8917820Z Elapsed time for package ace:x64-osx: 2.73 min
2020-04-02T00:00:24.8918000Z 
2020-04-02T00:00:24.8918410Z Triplet: x64-osx
2020-04-02T00:00:24.8918590Z Total elapsed time: 2.73 min
2020-04-02T00:00:24.8918890Z RESULTS
2020-04-02T00:00:24.8919320Z     ace:x64-osx: BUILD_FAILED: 2.73 min
```

jwillemsen commented:
> > Are all patches still required with TAO 2.5.8? Some got integrated already as far as I remember.
> 
> I did remove them, just two were still necessary. I will look into it again

The patches are not applied but still list of the changes, just remove them again from git I would like to suggest

bold84 committed:
Removed unnecessary patch files and other files

jwillemsen commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
-if("zlib" IN_LIST FEATURES)
-    list(APPEND ACE_FEATURE_LIST "zlib=1")
+if("xml" IN_LIST FEATURES)
+    list(APPEND ACE_FEATURE_LIST "xml=1")
```
There is no xml feature, should be xerces3

ordisbold replied:
Done

ordisbold commented:
@jwillemsen 
I have another port file for axcioma... but it builds ACE and TAO as well.
I didn't yet figure out how to use ACE/TAO from a separate build and I don't know if that would be a good idea either.

Any ideas on how we can make both coexist or how to combine them?

In my opinion it would be better to have an axcioma port with features "ace, tao, taox11, ..., ..." and make brix11 able to only build these targets if desired.

bold84 committed:
renamed feature xml to xerces-c

ordisbold commented:
And maybe it's possible to introduce some alias package sort of thing... so, if someone wants to install ace, it would actually install axcioma[ace].

ordisbold commented:
And another issue... is there something like feature dependencies?

xerces-c has an xmlch_wchar option.
If ace is built with wchar support and xerces-c support, shouldn't then be built with implicitly xmlch_wchar enabled?

jwillemsen commented:
Yes, I thought that worked, for xerces you need to pass xerces3=1 to mpc, not xml=1

bold84 committed:
reverted last change and replaced MPC feature xml with xerces3

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+###################################################
+
+# see https://htmlpreview.github.io/?https://github.com/DOCGroup/ACE_TAO/blob/master/ACE/ACE-INSTALL.html
+if(NOT VCPKG_CMAKE_SYSTEM_NAME)
```
Please replace it with
```
if (VCPKG_TARGET_IS_WINDOWS)
...
elseif (VCPKG_TARGET_IS_LINUX)
...
elseif(VCPKG_TARGET_IS_OSX)
...
elseif (VCPKG_TARGET_IS_UWP)
```

ordisbold replied:
It's using vcpkg_download_distfile now...

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+    file(WRITE ${ACE_SOURCE_PATH}/config.h "#include \"ace/config-windows.h\"\n#define ACE_NO_INLINE")
+endif()
+
+if((NOT VCPKG_CMAKE_SYSTEM_NAME) OR ("WindowsStore" STREQUAL VCPKG_CMAKE_SYSTEM_NAME))
```
Please replace it with `if (VCPKG_TARGET_IS_WINDOWS)`

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+  else()
+    set(SOLUTION_TYPE vc14)
+  endif()
+elseif((VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux") OR (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin"))
```
Please replaice it with `if (VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)`

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+###################################################
+
+# Build for Windows
+if((NOT VCPKG_CMAKE_SYSTEM_NAME) OR ("WindowsStore" STREQUAL VCPKG_CMAKE_SYSTEM_NAME))
```
Please replace it with `if (VCPKG_TARGET_IS_WINDOWS)`

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+# Build for Windows
+if((NOT VCPKG_CMAKE_SYSTEM_NAME) OR ("WindowsStore" STREQUAL VCPKG_CMAKE_SYSTEM_NAME))
+    vcpkg_build_msbuild(PROJECT_PATH "${WORKING_DIR}/${WORKSPACE}.sln" PLATFORM ${MSBUILD_PLATFORM} OPTIONS /m USE_VCPKG_INTEGRATION)
+elseif(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```
Please replaice it with `if (VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)`

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+if((NOT VCPKG_CMAKE_SYSTEM_NAME) OR ("WindowsStore" STREQUAL VCPKG_CMAKE_SYSTEM_NAME))
+    vcpkg_build_msbuild(PROJECT_PATH "${WORKING_DIR}/${WORKSPACE}.sln" PLATFORM ${MSBUILD_PLATFORM} OPTIONS /m USE_VCPKG_INTEGRATION)
+elseif(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux" OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  FIND_PROGRAM(MAKE make)
```
I think we should use `vcpkg_*_make`, see documentation:
[vcpkg_configure_make](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_make.md) / [vcpkg_build_make](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_build_make.md) / [vcpkg_install_make](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_make.md)

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+#
+###################################################
+
+if(NOT VCPKG_CMAKE_SYSTEM_NAME)
```
Please replace it with `if (VCPKG_TARGET_IS_WINDOWS)`

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
-endif()
-
-if(VCPKG_TARGET_IS_LINUX)
+elseif(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
Please replace it with `if (VCPKG_TARGET_IS_LINUX)`

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+  set(DLL_RELEASE_SUFFIX)
+  set(DLL_DEBUG_SUFFIX)
+  set(LIB_PREFIX lib)
+elseif(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```
Please replace it with `if (VCPKG_TARGET_IS_OSX)`

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+###################################################
+
+if(NOT VCPKG_CMAKE_SYSTEM_NAME)
   set(LIB_RELEASE_SUFFIX .lib)
```
These vars are already defined in [vcpkg_common_definitions](https://github.com/microsoft/vcpkg/blob/master/scripts/cmake/vcpkg_common_definitions.cmake).
Eg: `VCPKG_TARGET_STATIC_LIBRARY_SUFFIX` / `VCPKG_TARGET_SHARED_LIBRARY_SUFFIX`.

ordisbold replied:
Yep, but they don't handle static and debug suffixes on Windows, so I don't see any benefit of changing what already works and is very likely to work for the next decade.

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,323 @@
+# Install executables
+function(install_tao_executables SOURCE_PATH EXE_FILE)
+    set(EXECUTABLE_SUFFIX ".exe")
+    if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
Please replace it with `if (VCPKG_TARGET_IS_LINUX)`

bold84 committed:
Added VCPKG_TARGET_IS_*

bold84 committed:
replaced vcpkg_execute_required_process with vcpkg_build_make

jwillemsen commented:
> @jwillemsen
> I have another port file for axcioma... but it builds ACE and TAO as well.
> I didn't yet figure out how to use ACE/TAO from a separate build and I don't know if that would be a good idea either.
> 
> Any ideas on how we can make both coexist or how to combine them?
> 
> In my opinion it would be better to have an axcioma port with features "ace, tao, taox11, ..., ..." and make brix11 able to only build these targets if desired.

To start with I think you should let axcioma also generate the configuration files and build ACE/TAO. brix11 sets various flags which knowledge we don't want to duplicate again

ordisbold commented:
> > @jwillemsen
> > I have another port file for axcioma... but it builds ACE and TAO as well.
> > I didn't yet figure out how to use ACE/TAO from a separate build and I don't know if that would be a good idea either.
> > Any ideas on how we can make both coexist or how to combine them?
> > In my opinion it would be better to have an axcioma port with features "ace, tao, taox11, ..., ..." and make brix11 able to only build these targets if desired.
> 
> To start with I think you should let axcioma also generate the configuration files and build ACE/TAO. brix11 sets various flags which knowledge we don't want to duplicate again

Alright... I will create another PR after this one has been merged.

jwillemsen commented:
Looks linux/windows CI builds failed after the most recent changes

jwillemsen commented:
> Alright... I will create another PR after this one has been merged.

Just tag me when you have made that PR @ordisbold

bold84 committed:
replaced xml with xerces-c

bold84 committed:
disabled UWP builds

bold84 committed:
Removed Qt5 support

bold84 committed:
Don't support UWP

bold84 committed:
Don't support linux and expect such build to fail

ordisbold commented:
I don't understand why it's trying to build qt5-tools on Linux.

.\vcpkg.exe depend-info ace[*]
openssl-windows:
bzip2:
openssl: openssl-windows
xerces-c:
zlib:
ace[wchar, bzip2, xerces-c, ssl, tao, zlib]: bzip2, openssl, xerces-c, zlib

ordisbold commented:
I will focus on the axcioma based build... And I will not try to support all platforms and features immediately. :-D

JackBoosY commented:
@ordisbold Well, `qt5-tools` issue has been resolved. Could you reopen this PR again?

ordisbold commented:
> @ordisbold Well, `qt5-tools` issue has been resolved. Could you reopen this PR again?

I don't understand why it even required qt5-tools. I removed everything qt5 and it still failed due to an issue in the qt5 package. 🤔

I reopened it. 

But there's a problem regarding this port. 

TAO is based on ACE and it is not advisable to separate the build process of these libraries and make TAO a separate port. TAO only supports the decades old IDL-to-C++ mapping, which doesn't support the STL, because the STL wasn't part of the C++ standard when the IDL-to-C++ mapping was defined. The "old" IDL-to-C++ mapping is hard to use and developing with it is error-prone, especially for beginners. 

Renedy IT has crated a new IDL-to-C++11 mapping. It is now part of the official IDL to language specifications by the OMG. 

Remedy IT has also implemened the aforementioned IDL-to-C++11 mapping based on TAO and recently released it with other relevant components (i.e. LwCCM) as open source software. The suite is called AXCIOMA and  just the implementation of the new IDL-to-C++11 mapping is called TAOX11. 

Separating so if this is a can of worms I don't want to open. 

The question is: how to represent this in vcpkg? The AXCIOMA build process appears to be complex but stable. It implicitly downloads ACE and TAO and builds these libraries with specific settings that the library depends on.

JackBoosY commented:
/azp run

bold84 committed:
x86 -> Win32

bold84 committed:
"re-enable" linux build

bold84 committed:
working dir for linux build

bold84 committed:
removed nonsense argument.

ordisbold commented:
_VCPKG_MAKE_GENERATOR in vcpkg_build_make.cmake seems to be undefined. That's why the Linux build now fails :-D

Where is _VCPKG_MAKE_GENERATOR  set?

bold84 committed:
added vcpkg_configure_make

bold84 committed:
handle executable file extensions correctly

corrected platform specific library suffixes for libraries needed during runtime of tao_idl
temporarily disable Linux builds again :-P

JackBoosY commented:
@ordisbold `_VCPKG_MAKE_GENERATOR` is automatically set in `vcpkg_configure_make`.

JackBoosY commented on ports/ace/portfile.cmake:
```diff
@@ -1,244 +1,309 @@
+if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
+    vcpkg_build_msbuild(PROJECT_PATH "${WORKING_DIR}/${WORKSPACE}.sln" USE_VCPKG_INTEGRATION)
+elseif(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
+  vcpkg_build_make(SOURCE_PATH ${SOURCE_PATH})
```
You need to call `vcpkg_configure_make` first. See [example](https://github.com/microsoft/vcpkg/blob/19baa0d9b3be740fc107131759f89ce12b0038ac/ports/libmesh/portfile.cmake), [documentation](https://github.com/microsoft/vcpkg/blob/c0d22c88ea7638d1b74339f9e9adfd37b0f525ed/docs/maintainers/vcpkg_configure_make.md).

jwillemsen reviewed (changes requested):
Before this PR linux worked as far as I know, now it is disabled, personally I don't like to remove a platform again

JackBoosY reviewed (changes requested):
```
-- Applying patch /home/vcpkg/ports/ace/arm_prevent_amd64_definition.patch
CMake Error at scripts/cmake/vcpkg_apply_patches.cmake:51 (message):
  Applying patch failed.  Checking patch ace/config-win32-common.h...

  error: while searching for:

  #if defined (_WIN64) || defined (WIN64)

  # define ACE_WIN64



  // MPC template adds _AMD64_ but user projects not generated using MPC

  // may want to use _AMD64_ as well.  Ensure it's there in all cases.

  # ifndef _AMD64_

  # define _AMD64_

  # endif



  // Use 64-bit file offsets by default in the WIN64 case, similar to

  // what 64-bit UNIX systems do.

  //



  error: patch failed: ace/config-win32-common.h:39

  error: ace/config-win32-common.h: patch does not apply

Call Stack (most recent call first):
  ports/ace/portfile.cmake:42 (vcpkg_apply_patches)
  scripts/ports.cmake:90 (include)
```

bold84 committed:
Removed arm_prevent_amd64_definition.patch

ordisbold commented:
> Before this PR linux worked as far as I know, now it is disabled, personally I don't like to remove a platform again

Okay, but this PR took much longer than I anticipated. This was certainly due to my lack of knowledge of vcpkg and it's rules for ports but also because I intented to support all platforms right away. 

So, at the moment I want to only support Windows (I never thought I would ever write this :-P) and add other platforms in separate PRs. Otherwise I fear there will always be new issues, like the qt5-tools issue yesterday, which came and went away, even though I have removed all dependencies on qt5. 

I have already setup a CentOS VM yesterday to test Linux builds with it.

ordisbold commented:
I won't invest more time in this PR. 

I wanted to share the addition of TAO which I have been using here for a while and I didn't expect that the PR consumes more time than the addition of TAO to the port.

So, I don't require it to be merged.

jwillemsen commented:
Regret to see this closed, would be nice to have tao on vcpkg
